tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+	'Spontaneous', 
+	'Prepared'
+);
+
+CREATE TYPE IF NOT EXISTS ComponentType AS ENUM (
+	'Somatic',
+	'Material',
+	'Verbal'
+);
+
+CREATE TYPE IF NOT EXISTS MagicSchool AS ENUM (
+	'Abjuration', 
+	'Conjuration', 
+	'Divination', 
+	'Enchantment', 
+	'Evocation', 
+	'Illusion', 
+	'Necromancy', 
+	'Transmutation'
+);
+
+CREATE TYPE IF NOT EXISTS SpellRange AS ENUM (
+	'Personal',
+	'Touch',
+	'Close',
+	'Medium',
+	'Long',
+	'Unlimited'
+);
+
+CREATE TYPE IF NOT EXISTS WeaponClass AS ENUM (
+	'Axes', 
+	'Heavy Blades', 
+	'Light Blades', 
+	'Bows', 
+	'Close', 
+	'Crossbows', 
+	'Double', 
+	'Firearms', 
+	'Flails', 
+	'Hammers', 
+	'Monk', 
+	'Natural', 
+	'Polearms', 
+	'Siege Engines', 
+	'Spears', 
+	'Thrown', 
+	'Tribal'
+);
+
+CREATE TYPE IF NOT EXISTS ArmorClass AS ENUM (
+	'Light', 
+	'Medium', 
+	'Heavy'
+);
+
+CREATE TYPE IF NOT EXISTS EquipmentSlot AS ENUM (
+	'None',
+	'Armor', 
+	'Belts', 
+	'Body', 
+	'Chest', 
+	'Eyes', 
+	'Feet', 
+	'Hands', 
+	'Head', 
+	'Headband', 
+	'Neck', 
+	'Ring (left)', 
+	'Ring (right)', 
+	'Shield', 
+	'Shoulders', 
+	'Wrist'
+);
+
+CREATE TYPE IF NOT EXISTS DamageType AS ENUM (
+	'Bludgeoning', 
+	'Slashing', 
+	'Piercing', 
+	'Energy', 
+	'Acid', 
+	'Fire', 
+	'Electricity', 
+	'Cold', 
+	'Sonic', 
+	'Positive', 
+	'Negative', 
+	'Nonlethal'
+);
+
+--table declarations
+--user creds table - also used for hashing info
+CREATE TABLE IF NOT EXISTS Users (
+	id     			UUID PRIMARY KEY,
+
+	email			TEXT 
+		NOT NULL,
+        username       		TEXT 
+		NOT NULL,
+	nickname		TEXT 
+		NOT NULL,
+        is_admin    		BOOL 
+		NOT NULL,
+
+        pass_hash   		BYTEA 
+		NOT NULL,
+        salt        		BYTEA 
+		NOT NULL,
+
+        time_cost   		INT 
+		NOT NULL,
+        memory      		INT 
+		NOT NULL,
+        threads     		INT 
+		NOT NULL
+);
+
+--Character overview table
+CREATE TABLE IF NOT EXISTS Characters (
+        char_id         	UUID PRIMARY KEY,
+
+	id			UUID REFERENCES Users 
+		NOT NULL,
+        race_id         	UUID REFERENCES Races 
+		NOT NULL,
+        deity_id        	UUID REFERENCES Deities 
+		NOT NULL,
+
+        name            	TEXT 
+		NOT NULL,
+        age             	INT 
+		NOT NULL 
+		CHECK (age > 0),
+        gender          	Gender,
+        alignment       	Alignment 
+		NOT NULL,
+        backstory       	TEXT,
+        height          	INT,
+        weight          	INT,
+        size            	Size 
+		NOT NULL,
+
+        strength        	INT 
+		NOT NULL 
+		CHECK (strength >= 0 AND strength <= 20),
+        dexterity       	INT 
+		NOT NULL 
+		CHECK (dexterity >= 0 AND dexterity <= 20),
+        constitution    	INT 
+		NOT NULL
+		CHECK (constitution >= 0 AND constitution <= 20),
+        intelligence    	INT 
+		NOT NULL
+		CHECK (intelligence >= 0 AND intelligence <= 20),
+        wisdom          	INT
+		NOT NULL
+		CHECK (wisdom >= 0 AND wisdom <= 20),
+        charisma        	INT
+		NOT NULL
+		CHECK (charisma >= 0 AND charisma <= 20),
```

Comment:
Are you sure stats can *never* be greater than 20?

Reply:
The stored stat can never be greater than 20, but the _computed value_ can be greater than 20. If your Strength is at 20, you can't put another level into it - but you can drink a Potion of Strength for a +2 to the stat.